### PR TITLE
add resp hook and write retry cases

### PR DESCRIPTION
**Pull request by GTwhy**

### CODE REVIEW

File: src/case/sim_write_partial_retry.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - name: remote_write
+    len: 1024
+  - name: barrier
+
+side_2:
+  - name: prepare
+  - name: connect_qp
+    mtu: 256
+  - name: set_hook
+    hook_type: 2
```
Comment (rogercloud) [inline src/case/sim_write_partial_retry.yaml]: Can we make hook type more readable?

**Review by rogercloud — COMMENTED** [review]
Here we have three hooks. But they're called in different stages of each processes, for instance, send hook is called before any other operations. Can we comment the position and explain why we add hooks there?

File: src/case/sim_write_partial_retry.yaml (hunk 1 above)
Comment (GTwhy) [reply src/case/sim_write_partial_retry.yaml]: Good advice ; )

**Comment by GTwhy** [comment]
> Here we have three hooks. But they're called in different stages of each processes, for instance, send hook is called before any other operations. Can we comment the position and explain why we add hooks there?

Yes, I should add more comments

**Review by rogercloud — APPROVED** [review]

**Pull request merged**